src/NuGet.Core/NuGet.Packaging/PackageExtraction/ZipArchiveExtensions.cs

```diff
@@ -67,25 +68,65 @@ public static string SaveAsFile(this ZipArchiveEntry entry, string fileFullPath,
 
         public static void UpdateFileTimeFromEntry(this ZipArchiveEntry entry, string fileFullPath, ILogger logger)
         {
-            var attr = File.GetAttributes(fileFullPath);
+            Testable.Default.UpdateFileTimeFromEntry(entry, fileFullPath, logger);
+        }
+
+        internal class Testable
+        {
+            public static Testable Default { get; } = new Testable(EnvironmentVariableWrapper.Instance);
 
-            if (!attr.HasFlag(FileAttributes.Directory) &&
-                entry.LastWriteTime.DateTime != DateTime.MinValue && // Ignore invalid times
-                entry.LastWriteTime.UtcDateTime <= DateTime.UtcNow) // Ignore future times
+            internal Testable(IEnvironmentVariableReader environmentVariableReader)
             {
-                try
+                _updateFileTimeFromEntryMaxRetries = 9;
+                string value = environmentVariableReader.GetEnvironmentVariable("NUGET_UpdateFileTime_MaxRetries");
+                if (int.TryParse(value, out int maxRetries) && maxRetries > 0)
```

Comment:
``` suggestion
                if (int.TryParse(value, out int maxRetries) && maxRetries > _updateFileTimeFromEntryMaxRetries)
```

The reason is `_updateFileTimeFromEntryMaxRetries` value is the default minimum value which we are considering for retries.

Reply:
Why do you believe it's the minimum?  I don't know why someone would want to set it to only 1 or 2, but if they feel like it's desirable, why would we want to prevent them?

Reply:
Great question. In general, customers would not be familiar with the new environment variable until they hit this problem. Then customers would read the docs and try to add the environment variable with a value greater than the default value and run the restore again. I am wondering if customers are confident that 1 or 2 retries would work then why bother adding a new environment variable because the current loop terminates when the file timestamp is set with an upper bound of 9 retries. I am fine with the existing implementation and happy to resolve this comment with no further action required.